src/analysis/calculus/specific_functions.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2020 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import analysis.calculus.extend_deriv analysis.calculus.iterated_deriv analysis.complex.exponential
+
+/-!
+# Smoothness of specific functions
+
+The real function `exp_neg_inv_glue` given by `x ↦ exp (-1/x)` for `x > 0` and `0`
+for `x ≤ 0` is a basic building block to construct smooth partitions of unity. We prove that it
+is `C^∞` in `exp_neg_inv_glue_smooth`.
+-/
+
+noncomputable theory
+open_locale classical topological_space
+
+namespace partition_of_unity
+open polynomial real filter set
+
+/-- `exp_neg_inv_glue` is the real function given by `x ↦ exp (-1/x)` for `x > 0` and `0`
+for `x ≤ 0`. is a basic building block to construct smooth partitions of unity. Its main property
+is that it vanishes for `x ≤ 0`, it is positive for `x > 0`, and the junction between the two
+behaviors is flat enough to retain smoothness. The fact that this function is `C^∞` is proved in
+`exp_neg_inv_glue_smooth`. -/
+def exp_neg_inv_glue (x : ℝ) : ℝ := if x ≤ 0 then 0 else exp (-x⁻¹)
+
+/-- Our goal is to prove that `exp_neg_inv_glue` is `C^∞`. For this, we compute its successive
+derivatives for `x > 0`. The `n`-th derivative is of the form `P_n(x) exp(-1/x) / x^(2 n)`,
+where `P_n` is computed inductively. We define `exp_neg_inv_aux_poly n` to be this `P_n`. -/
+noncomputable def exp_neg_inv_aux_poly : ℕ → polynomial ℝ
+| 0 := 1
+| (n+1) := X^2 * (exp_neg_inv_aux_poly n).derivative  + (1 - C (2 * n) * X) * (exp_neg_inv_aux_poly n)
+
+/-- Formula for the `n`-th derivative of `exp_neg_inv_glue`. -/
+def exp_neg_inv_glue_aux (n : ℕ) (x : ℝ) : ℝ :=
+if x ≤ 0 then 0 else (exp_neg_inv_aux_poly n).eval x * exp (-x⁻¹) / x^(2 * n)
+
+/-- The `0`-th auxiliary function `exp_neg_inv_glue_aux 0` coincides with `exp_neg_inv_glue`, by
+definition. -/
+lemma exp_neg_inv_glue_eq_exp_neg_inv_glue_aux_zero :
```

Comment:
How about
```suggestion
lemma exp_neg_inv_glue_aux_zero_eq :
```
and swap the direction, so that it is in simp form?

Reply:
Done